app/src/main/java/org/mozilla/focus/utils/GifImageView.java

```diff
@@ -0,0 +1,100 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Source taken from https://stackoverflow.com/questions/42898968/adding-gif-in-android-studio
+
+package org.mozilla.focus.utils;
+
+import android.content.Context;
+import android.view.View;
+import java.io.InputStream;
+import android.graphics.Movie;
+import android.graphics.Canvas;
+import android.util.AttributeSet;
+import android.os.SystemClock;
+import android.net.Uri;
+import java.io.FileNotFoundException;
+import android.util.Log;
+
+public class GifImageView extends View {
+
+    private InputStream mInputStream;
+    private Movie mMovie;
+    private int mWidth, mHeight;
+    private long mStart;
+    private Context mContext;
+
+    public GifImageView(Context context) {
+        super(context);
+        this.mContext = context;
+    }
+
+    public GifImageView(Context context, AttributeSet attrs) {
+        this(context, attrs, 0);
+    }
+
+    public GifImageView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        this.mContext = context;
+        if (attrs.getAttributeName(1).equals("background")) {
+            int id = Integer.parseInt(attrs.getAttributeValue(1).substring(1));
+            setGifImageResource(id);
+        }
+    }
+
+
+    private void init() {
+        setFocusable(true);
+        mMovie = Movie.decodeStream(mInputStream);
```

Comment:
nit: this inputStream needs to be closed once it's used:
```java
try {
    mMovie = Movie.decodeStream(mInputStream);
} finally {
    mInputStream.close();
}
```

In this case, mInputStream doesn't need to be a member variable (which would make it easy to re-use it when it's already closed) – it you can pass it as an arg to init.